src/crimson/osd/osd_operations/compound_peering_request.cc

```diff
@@ -160,17 +160,16 @@ seastar::future<> CompoundPeeringRequest::start()
 	boost::static_pointer_cast<MOSDPGCreate2>(m));
     }());
 
-  add_blocker(blocker.get());
   IRef ref = this;
   logger().info("{}: about to fork future", *this);
-  return state->promise.get_future().then(
-    [this, blocker=std::move(blocker)](auto &&ctx) {
-      clear_blocker(blocker.get());
-      logger().info("{}: sub events complete", *this);
-      return osd.get_shard_services().dispatch_context_messages(std::move(ctx));
-    }).then([this, ref=std::move(ref)] {
-      logger().info("{}: complete", *this);
-    });
+  return with_blocking_future(
```

Comment:
could you note in the commit message that it's a cleanup (IIUC)?

Reply:
Commit message added.